src/libraries/Microsoft.Bcl.Cryptography/src/System/Security/Cryptography/Shake256.cs

```diff
@@ -4,35 +4,196 @@
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Text;
+using System.Threading;
+using Internal.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+using NTSTATUS=Interop.BCrypt.NTSTATUS;
 
 namespace System.Security.Cryptography
 {
 #if !NET
     internal sealed class Shake256 : IDisposable
     {
+#if NETFRAMEWORK
+        private volatile int _concurrentCount;
+        private readonly SafeBCryptHashHandle _hashHandle;
+
+        internal Shake256()
+            : this(CreateHashHandle())
+        {
+        }
+
+        internal Shake256(SafeBCryptHashHandle hashHandle)
+        {
+            Debug.Assert(hashHandle != null);
+            _hashHandle = hashHandle;
+        }
+
+        internal void AppendData(ReadOnlySpan<byte> data)
+        {
+            if (data.IsEmpty)
+            {
+                return;
+            }
+
+            int count = Interlocked.Increment(ref _concurrentCount);
```

Comment:
This has slightly different concurrency semantics than the in-box one. This limits total concurrency around the using the instance, whereas the `ConcurrencyBlock` is just around individual callsites (since that is all the concurrency limiter required us to do).

I think having the concurrency blocking behavior consistent is perhaps a worthy goal.

Reply:
In other words, just pull in `ConcurrencyBlock.cs` :-D.

Reply:
Nevermind I misread this. All good.

Reply:
I started by copying over ConcurrencyBlock. VS gave me a warning that it used features unavailable on netfx, so I did this version instead.